### PR TITLE
Improve Error reporting from the matrix connector and database

### PR DESCRIPTION
**Pull request by Cadair**
This resolves the first issue in #1529

### CODE REVIEW

**Comment by Cadair** [comment]
I think this is the last major blocker to a release.

**Comment by SolarDrew** [comment]
LGTM :+1:

**Review by SolarDrew — APPROVED** [review]

File: opsdroid/connector/matrix/connector.py [hunk 1]
```diff
@@ -264,6 +277,8 @@ async def _parse_sync_response(self, response):
                     if event.sender != self.mxid:
                         if event.source["type"] == "m.room.member":
                             event.source["content"] = event.content
+                        if isinstance(event, nio.MegolmEvent):
+                            _LOGGER.error(f"Failed to decrypt event {event}")
```
Comment (FabioRosado) [inline opsdroid/connector/matrix/connector.py:281]: It seems that this is the only thing that is making cov complain. Can you write a silly test to get that log line covered?

Comment (Cadair) [reply opsdroid/connector/matrix/connector.py:281]: ffs codecov

Comment (Cadair) [reply opsdroid/connector/matrix/connector.py:281]:
```suggestion
                            _LOGGER.error(f"Failed to decrypt event {event}")  # pragma: no cover
```

**Comment by jacobtomlinson** [comment]
What is left to do here (other than fix the linting)?

**Comment by jacobtomlinson** [comment]
🎉

**Pull request merged**